src/backend/distributed/utils/node_metadata.c

```diff
@@ -835,8 +872,16 @@ GetNodeTuple(char *nodeName, int32 nodePort)
 	heapTuple = systable_getnext(scanDescriptor);
 	if (!HeapTupleIsValid(heapTuple))
 	{
-		ereport(ERROR, (errmsg("could not find valid entry for node \"%s:%d\"",
-							   nodeName, nodePort)));
+		if (raiseError)
+		{
+			ereport(ERROR, (errmsg("could not find valid entry for node \"%s:%d\"",
```

Comment:
That's probably better, I'm not happy that we've now duplicated the error message but I think you're right that the code is cleaner now.